nixos/modules/services/cluster/kubernetes/apiserver.nix

```diff
@@ -421,8 +459,32 @@ in
         };
       };
 
-    })
+    }))
+    {
+      systemd.targets.kube-apiserver-online = {
+        wantedBy = [ "kubernetes.target" ];
+        before = [ "kubernetes.target" ];
+      };
 
+      systemd.services.kube-apiserver-online = mkIf top.flannel.enable {
```

Comment:
Hmm, i see. The mkIf currently is there because the mkWaitCurl is using the certifificates of the flannel client to check the apiserver /healthz path. 
I decided to use the flannel certs because it seemed to me that it will always be present on the master and nodes.

So maybe on the master the mkWaitCurl could be called with the cluster-admin cert, and maybe this will work with the kubelet-client cert on the nodes (not sure if this is working because the node somehow is not present at this point, but will try)

Reply:
Resolved in https://github.com/NixOS/nixpkgs/pull/56789/commits/7323b77435f69362b0b4cc7edcb0915e9ab1ff48#diff-8d468cb36655335308a8eb4c3999b59aR372 by either using the `clusterAdmin` cert or the `kubelet` cert.

Reply:
See my comment here: https://github.com/NixOS/nixpkgs/commit/7323b77435f69362b0b4cc7edcb0915e9ab1ff48#r32635029